go/pkg/client/client.go

```diff
@@ -76,19 +85,31 @@ func NewClient(ctx context.Context, host string, port string, options ...ClientO
 
 	client.ticketFact = newTicketFactory()
 
-	client.sessionStub, err = newSessionStub(ctx, client)
+	client.flightStub, err = newFlightStub(client, host, port)
 	if err != nil {
 		client.Close()
```

Comment:
Defer doesn't work, since the client is being returned.  We don't want to close it in that case.  You only want to close in the error cases.  For those error cases, the client may be partially built, so the partially built client needs to be closed. ... which is going to make me check that there are not NPEs for a partial close.

Reply:
Would it work to write code like this at the start
```
	workingClient := &Client{grpcChannel: grpcChannel, isOpen: true}
        defer {
            if workingClient != nil
                 workingClient.Close()
        }
	workingClient.suppressTableLeakWarning = opts.suppressTableLeakWarning
        etc etc
```

and then at the bottom

```
         client = workingClient
         workingClient = nil
         return client, nil
```